test/layeredmaterial_unit_test.js

```diff
@@ -0,0 +1,44 @@
+import LayerUpdateState from '../src/Core/Layer/LayerUpdateState';
+import { updateLayeredMaterialNodeImagery } from '../src/Process/LayeredMaterialNodeProcessing';
+/* global describe, it */
+
+const assert = require('assert');
+
+describe('material state vs layer state', function () {
+    let opacity = 0.0;
+
+    const node = {
+        parent: { },
+        layerUpdateState: {
+            test: new LayerUpdateState(),
+        },
+        material: {
```

Comment:
Food for thought, what you did here is actually a cheap version of sinon.js (or equivalent). It's a sort of spy. I'm not saying we should introduce this dependency, but let's keep it in mind if this use case becomes frequent.